include/instantiate.h

```diff
@@ -232,7 +232,118 @@ namespace quda
 #endif
     } else if (field.Precision() == QUDA_QUARTER_PRECISION) {
 #if QUDA_PRECISION & 1
-      instantiate<Apply, char>(field, args...);
+      instantiate<Apply, int8_t>(field, args...);
+#else
+      errorQuda("QUDA_PRECISION=%d does not enable quarter precision", QUDA_PRECISION);
+#endif
+    } else {
+      errorQuda("Unsupported precision %d\n", field.Precision());
+    }
+  }
+
+  /**
+     @brief The instantiatePrecision function is used to instantiate
+     the precision
+     @param[in] field LatticeField we wish to instantiate
+     @param[in,out] args Any additional arguments required for the
+     computation at hand
+  */
+  template <template <typename> class Apply, typename F, typename... Args>
+  constexpr void instantiatePrecision(F &field, Args &&... args)
+  {
+    if (field.Precision() == QUDA_DOUBLE_PRECISION) {
+      // always instantiate double precision
```

Comment:
Why is double always being instantiated?

Reply:
This is a special case template: for the interface we need to always instantiate double precision for the copy in/out kernels.

Reply:
Understood, please add that to the doxygen then